app/api/uploadURL.ts

```diff
@@ -0,0 +1,32 @@
+import { Storage } from '@google-cloud/storage'
+import { BlitzApiHandler } from 'blitz'
+
+import {
+  gcpStorageBucketName,
+  gcpStorageClientEmail,
+  gcpStoragePrivateKey,
+  gcpStorageProjectId,
+} from 'app/core/environment'
+
+const handler: BlitzApiHandler = async (req, res) => {
+  const gc = new Storage({
+    projectId: gcpStorageProjectId,
+    credentials: {
+      client_email: gcpStorageClientEmail,
+      private_key: gcpStoragePrivateKey,
+    },
+  })
+
+  const bucket = gc.bucket(gcpStorageBucketName)
+  const filename = req.query.file as string
+  const folder = req.query.dir as string
```

Comment:
File name and folder name shouldn't be a user input, else users can override other users' files. 

I propose a name format:
- folder name: /users/[userId]
- file name: randomly generated uuid
And have the server generate the file name and upload url when the user wants to upload a file.